src/MUIDataTable.js

```diff
@@ -259,52 +259,56 @@ class MUIDataTable extends React.Component {
     },
   };
 
-  state = {
-    activeColumn: null,
-    announceText: null,
-    count: 0,
-    columns: [],
-    expandedRows: {
-      data: [],
-      lookup: {},
-    },
-    data: [],
-    displayData: [],
-    filterData: [],
-    filterList: [],
-    page: 0,
-    previousSelectedRow: null,
-    rowsPerPage: 0,
-    searchProps: {},
-    searchText: null,
-    selectedRows: {
-      data: [],
-      lookup: {},
-    },
-    showResponsive: false,
-    sortOrder: {},
-  };
-
-  constructor() {
-    super();
+  constructor(props) {
+    super(props);
     this.tableRef = React.createRef();
     this.tableContent = React.createRef();
     this.draggableHeadCellRefs = {};
     this.resizeHeadCellRefs = {};
     this.timers = {};
     this.setHeadResizeable = () => {};
     this.updateDividers = () => {};
-  }
 
-  UNSAFE_componentWillMount() {
-    this.initializeTable(this.props);
+    let defaultState = {
+      activeColumn: null,
+      announceText: null,
+      count: 0,
+      columns: [],
+      expandedRows: {
+        data: [],
+        lookup: {},
+      },
+      data: [],
+      displayData: [],
+      filterData: [],
+      filterList: [],
+      page: 0,
+      previousSelectedRow: null,
+      rowsPerPage: 0,
+      searchProps: {},
+      searchText: null,
+      selectedRows: {
+        data: [],
+        lookup: {},
+      },
+      showResponsive: false,
+      sortOrder: {},
+    };
+
+    this.mergeDefaultOptions(props);
+    this.state = Object.assign(defaultState, this.getInitTableOptions());
+    this.setTableData = this.setTableData.bind(this);
+
+    this.setTableData(props, TABLE_LOAD.INITIAL, true, null, true);
   }
 
   componentDidMount() {
     this.setHeadResizeable(this.resizeHeadCellRefs, this.tableRef);
 
     // When we have a search, we must reset page to view it unless on serverSide since paging is handled by the user.
     if (this.props.options.searchText && !this.props.options.serverSide) this.setState({ page: 0 });
+
+    this.setTableInit('tableInitialized');
```

Comment:
But shouldn't this be the first thing to do?

```js
componentDidMount() {
  this.setTableInit('tableInitialized');

  ...
}
```

Reply:
To keep consistent, it probably should. I'll move it up.